### DISCUSSION TITLE
How to pass an object in params in routes?

### DISCUSSION THREAD

**Discussion by brunowilliang** [expo/router#502] [Q&A]
Guys, I need to pass an object in my routes, but it doesn't send it the way I want.

I'm going like this:
```tsx
  const handleTransaction = (item: TransactionsResponse) => {
    router.push({
      pathname: `/transactions/${item?.id}`, params: { item: item }
    });
  }
```

const params = useSearchParams();
console.log(params)

```shel
 LOG  {"id": "eu33d3uekkcq6dh", "initial": "true", "item": "[object Object]", "params": "[object Object]", "screen": "home", "state": "undefined"}
```

Can someone help me?

**Comment by wcastand** [comment]
i'd say you got 2 solutions:

First, you `JSON.stringify` your object and `JSON.parse` in the screen (i would not do that)

Second is you have in the url the id of the transaction so the "best" practice would be to query that transaction in the screen based on the id you pass in the url. Ideally you have a cache system that would avoid waiting for an other network call to get the item.

**Reply by likeSo** [reply]
But if we pass only id, we have to make networking again... The user experience is not ok...

**Reply by wcastand** [reply]
you should have a cache in your app for that, that will fetch in background but show you the cache data straight away so the user don't get a loading again but still get the latest data.

**Comment by likeSo** [comment]
Why can't the expo router just bring the original React Navigation feature back?
Even if we pass objects through `JSON.stringify`, the href string looks very awful
